### ISSUE TITLE
Standard command-line options to help swapping

### ISSUE THREAD

**Issue by rjelliffe** [Schematron/schematron-enhancement-proposals#77]
**Proposal**
A new non-normative annex should be provided. An implementation can support the following command-line options, which take an ID argument:

- Select/Deselect phase  F - one phase is active at a time, also allowing #DEFAULT or #ANY 
- Activate/Deactive pattern P
- Enable/Disable rule R  - id of rule. The rule context is still evaluated, except for disabled rules at the end of the pattern, but none of its assertions and reports are tested. So there is no change in the results of other non-disabled rules in the pattern.
- Enable/Disable assertion/report A  

Other options (for parameters) should be provided.  

The purpose of this to allow better swappability of different implementations. For example, so that a Schematron schema can generate a pop-up dialog box with checkboxes for all phases, patterns, rules and assertions to allow the user to enable or disable, and so that then this enabling/disabling can easily  be transferred to the engine.  (Whether implementers do it by a run-time test or compile-time test or by text pre-processing of the schema is implementation-specific.) 

These commands should be provided also as URNs, (for example, these might be properties set on a Java validator object factory.)

This will help new implementers know the kind of thing their implementation should provide, and especially may help allowing swapping out of one implementation for another.

**Comment by AndrewSales** [comment]
This is for implementers to decide, and doesn't belong in the standard, in my view.

**Comment by rjelliffe** [comment]
Yes it does belong in the body of a standard. It is presumably the thing that might be better in a TR rather than a non-nrmtve annex. And maybe there are not enough implementations to mean co-ordination is overkill.

**Comment by dmj** [comment]
I agree with @AndrewSales here: This is something that implementers should agree upon.
